src/main/java/org/folio/rest/impl/PrecedingSucceedingTitleAPI.java

```diff
@@ -68,6 +94,56 @@ public void putPrecedingSucceedingTitlesByPrecedingSucceedingTitleId(String prec
       PutPrecedingSucceedingTitlesByPrecedingSucceedingTitleIdResponse.class, asyncResultHandler);
   }
 
+  private boolean validatePrecedingSucceedingTitles(List<PrecedingSucceedingTitle> precedingSucceedingTitles,
+                                                    String instanceId,
+                                                    Handler<AsyncResult<Response>> asyncResultHandler) {
+    boolean areValidTitles = true;
+    for (PrecedingSucceedingTitle precedingSucceedingTitle : precedingSucceedingTitles) {
+      if (titleIsLinkedToInstanceId(precedingSucceedingTitle, instanceId)) {
+        var validationErrorMessage =
+          createValidationErrorMessage("precedingInstanceId or succeedingInstanceId", "",
+            String.format("The precedingInstanceId or succeedingInstanceId should contain instanceId [%s]", instanceId));
+        asyncResultHandler.handle(Future.succeededFuture(PutPrecedingSucceedingTitlesResponse
+          .respond422WithApplicationJson(validationErrorMessage)));
+        areValidTitles = false;
+      }
+    }
+    return areValidTitles;
+  }
+
+  private boolean titleIsLinkedToInstanceId(PrecedingSucceedingTitle precedingSucceedingTitle, String instanceId) {
+    return !instanceId.equals(precedingSucceedingTitle.getPrecedingInstanceId())
+      && !instanceId.equals(precedingSucceedingTitle.getSucceedingInstanceId());
```

Comment:
The more negations code has the more difficult to understand it is.
I suggest 
```
    return instanceId.equals(precedingSucceedingTitle.getPrecedingInstanceId())
      || instanceId.equals(precedingSucceedingTitle.getSucceedingInstanceId());
```
and using
```
if (! titleIsLinkedToInstanceId(precedingSucceedingTitle, instanceId))
```